adam_preprocessing/shape_stroke_extraction.py

```diff
@@ -738,6 +741,24 @@ def get_colors(self):
         # objects for those locations are (considered) distinct. I think that is the convention
         # here.
         img_seg = cv2.imread(self.path)
+        all_seg_pixels = np.reshape(img_seg, (-1, img_seg.shape[-1]))
+        all_unique_colors = np.unique(all_seg_pixels, axis=0)
+
+        # Ignore black (background) pixels
+        unique_colors = all_unique_colors[
+            np.where((all_unique_colors != [0, 0, 0]).any(axis=-1))
+        ]
+        # Array of boolean masks (one for each color), shape: (n_masks, height, width)
+        masks = np.all(
+            img_seg[np.newaxis, ...] ==
+            unique_colors[:, np.newaxis, np.newaxis, :], axis=-1
+        )
+        for i in range(unique_colors.shape[0]):
+            color = unique_colors[i, np.newaxis]
+            int_mask = masks[i, :, :, np.newaxis].astype(int)
+            color_mask = int_mask * color
```

Comment:
Could do `img_seg[masks[i, :, :, np.newaxis]]`? Does that work?

Reply:
Huh, that's annoying. I swear there should be an indexing way to do this but it's not too worth fixing. I'm okay with this as-is.

Not necessary to change, but I wanted to think about how to do it -- here's an idea. I think we don't need `masks` at all. We can do `color_mask = np.where(img_seg == unique_colors[i], img_seg, [0, 0, 0])`. I tested this in a Python console and it seems to work with a two-mask test image.

Reply:
Yep, this works!